content/blog/debugging-nodejs-programs-using-chrome/index.md

```diff
@@ -0,0 +1,308 @@
+>
+> This way, you should have the margins to add in a breakpoint where you'd like one beforehand.
+
+# Using The Debugging Tools {#using-debug-tools}
+
+Once your code runs through a breakpoint, this window should immediately raise to focus (even if it's in the background).
+
+![A breakpoint paused on line 7 at the JSON parsing line](breakpoint-paused.png)
+
+> If you don't see the `Console` tab at the bottom of the screen, as is shown here, you can bring it up by pressing the `Esc` key. This will allow you to do various interactions with your code that are outlined below
+
+Once you do so, you're in full control of your code and its state. You can:
+
+- _Inspect variable's values_ (either by highlighting the variable you're interested in, looking under the "scope" tab on the right sidebar, or using the `Console` tab to run inspection commands ala !!! [`console.table`](https://developer.mozilla.org/en-US/docs/Web/API/Console/table) or [`console.log`](https://developer.mozilla.org/en-US/docs/Web/API/Console/log))
+	![A screenshot of all three of the mentioned methods to inspect a variable's value](./inspect-variable-value.png)
+- _Change the value of a variable_
+  ![A screenshot of using the Console tab in order to change the value of a variable as you would any other JavaScript variable](./change-variable-value.png)
+- _Run arbitrary JavaScript commands_, similar to how a code playground might allow you to
+  ![A screenshot of indexing the body using "body.slice(0, 100)"](./arbitrary-js.png)
+
+## Running Through Lines {#running-through-lines}
+
+But that's not all! Once you make changes (or introspect the values), you're also able to control the flow of your application. For example, you may have noticed the following buttons in the debug window:
+
+![A red circle highlighting the "play" and "skip" buttons in the debugger](./breakpoint-paused-buttons.png)
+
+Both of these buttons allow you to control where your debugger moves next. _The button to the left_ is more of a "play/pause" button. Pressing this _will unpause your code and keep running it_ (with your variable changes in-tact) _until it hits the next breakpoint_. If this happens to be two lines down, then it will run the line in-between without pausing and then pause once it reached that next breakpoint.
+
+So, if we want to see what happens after the `body` JSON variable is parsed into a variable, we could press the "next" button to the right to get to that line of code and pause once again.
+
+![A screenshot of the JSON being parsed into a few variables with some console logs to prove it did really parse and run the line above it](./next-line.png)
+
+ Knowing this, let's move through the next few lines manually by pressing each item. The ran values of the variables as they're assigned should show up to the right of the code itself in a little yellow box; This should help you understand what each line of code is running and returning without `console.log`ging or otherwise manually.
```

Comment:
```suggestion
Knowing this, let's move through the next few lines manually by pressing each item. The ran values of the variables as they're assigned should show up to the right of the code itself in a little yellow box; This should help you understand what each line of code is running and returning without `console.log`ging or otherwise manually.
```